CONTRIBUTING.md

```diff
@@ -0,0 +1,108 @@
+
+Contributing code
+=================
+
+How to contribute
+-----------------
+
+The preferred way to contribute to Kymatio is to fork the
+[main repository](http://github.com/kymatio/kymatio/) on
+GitHub:
+
+1. Fork the [project repository](http://github.com/kymatio/kymatio):
+   click on the 'Fork' button near the top of the page. This creates
+   a copy of the code under your account on the GitHub server.
+
+2. Clone this copy to your local disk:
+
+          $ git clone git@github.com:YourLogin/kymatio.git
+          $ cd kymatio
+
+3. Remove any previously installed versions of Kymatio:
+
+          $ pip uninstall kymatio
+
+and install your local copy with testing dependencies:
+
+          $ pip install -e .[tests]
+
+4. Create a branch to hold your changes:
+
+          $ git checkout -b my-feature
+
+   and start making changes. Never work in the ``master`` branch.
+
+5. Work on this copy on your computer using Git to do the version
+   control. When you're done editing, do:
+
+          $ git add modified_files
+          $ git commit
+
+   to record your changes in Git, then push them to GitHub with:
+
+          $ git push -u origin my-feature
+
+Finally, go to the web page of the your fork of the librosa repo,
+and click 'Pull request' to send your changes to the maintainers for
+review. This will send an email to the committers.
+
+For further details on using Git for version control, we recommend you look
+up its [documentation](http://git-scm.com/documentation).
+
+
+
+Filing bugs
+-----------
+We use Github issues to track all bugs and feature requests. Feel free to
+open an issue if you have found a bug or wish to see a feature implemented.
+
+It is recommended to check that your issue complies with the
+following rules before submitting:
+
+-  Verify that your issue is not being currently addressed by other
+   [issues](https://github.com/kymatio/kymatio/issues?q=)
+   or [pull requests](https://github.com/kymatio/kymatio/pulls?q=).
+
+-  Please ensure all code snippets and error messages are formatted in
+   appropriate code blocks.
+   See [Creating and highlighting code blocks](https://help.github.com/articles/creating-and-highlighting-code-blocks).
+
+-  Please include your operating system type and version number, as well
+   as your Python, NumPy, SciPy, PyTorch, and Kymatio versions. This information
+   can be found by running the following code snippet:
+
+  ```python
+  import platform; print(platform.platform())
+  import sys; print("Python", sys.version)
+  import scipy; print("NumPy", scipy.__version__)
+  import scipy; print("SciPy", scipy.__version__)
+  import torch; print("PyTorch", torch.__version__)
+  import kymatio; print("Kymatio", kymatio.__version__)
+  ```
+
+- If your issue is related to GPU acceleration, please copy-paste the output
+  from the [environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
```

Comment:
Huh cool! What's that? :)

Reply:
it's a really complex but very cool script written by PyTorch people, which gives very detailed information on the environment, including versions of GCC compiler, GPU models and configuration, NVIDIA drivers, etc. it's really cool
The only drawback is that it takes a while to run (10-20 seconds at least)
in  the long term we could fork this script and make one for our own purposes of troubleshooting